app/helpers/application_helper/button/middleware_server_action.rb

```diff
@@ -1,7 +1,18 @@
 class ApplicationHelper::Button::MiddlewareServerAction < ApplicationHelper::Button::Basic
 
   def visible?
-    !@record.present? ||
-      (@record.try(:product) != 'Hawkular' && @record.try(:middleware_server).try(:product) != 'Hawkular')
+    @record.present? && !hawkular? && !immutable?
+  end
+
+  private
+
+  def immutable?
+    properties = @record.try(:properties)
+    properties.nil? ? false : properties['Immutable'] == 'true'
+  end
+
+  def hawkular?
+    @record.try(:product) == 'Hawkular' ||
+      @record.try(:middleware_server).try(:product) == 'Hawkular'
```

Comment:
It has been already merged, but it may be handy to have those methods also on the server entity itself